BASE/Test/Microsoft.ApplicationInsights.Test/Microsoft.ApplicationInsights.Tests/Extensibility/Implementation/Tracing/CoreEventSourceTest.cs

```diff
@@ -25,5 +26,30 @@ private static EventAttribute GetEventAttribute(string methodName)
             MethodInfo method = typeof(CoreEventSource).GetMethod(methodName);
             return method.GetCustomAttribute<EventAttribute>();
         }
+
+#if REDFIELD
+        /// <summary>
+        /// This is a sanitiy check.
+        /// The 'Redfield' compilation flag should switch the name of EventSource class.
+        /// Devs can review the test log and confirm that this test runs and passes.
+        /// This will serve as a verification that the Redfield compilation flag worked as expected.
+        /// 
+        /// To run this test:
+        /// dotnet build /p:Redfield=True ".\dotnet\BASE\Microsoft.ApplicationInsights.sln"
```

Comment:
Will this test be covered by GitHub Action?

Reply:
Yes. We run a package against pipeline with compilation flag and that validates it. 
I validated this test in this PR itself. If you look at the commit `Remove hard-coded Redfield property` the tests ran successfully before that commit should include this test.

Reply:
now that i've thought about it, i think Sean is right. lets introduce a simple github action to sanity check this compilation. @rajkumar-rangaraj ping me on Monday, i have another branch with a github compatible yml that we can repurpose for this.